### PR TITLE
Developer docs

### PR DESCRIPTION
**Pull request by balos1**
Also adds instructions for updating answers in the developer docs.

RTD preview: https://sundials.readthedocs.io/en/feature-udpating-answers-instructions/

### CODE REVIEW

**Review by drreynolds — DISMISSED** [review]

File: doc/superbuild/source/developers/index.rst [hunk 1]
```diff
@@ -12,22 +12,29 @@
...
-   introduction/index
+   History
    getting_started/index
    style_guide/index
    development/index
    testing/index
    benchmarks/index
    pull_requests/index
    releases/index
-   packages/index
```
Comment (gardner48) [inline doc/superbuild/source/developers/index.rst:31]: Looks like this got lost in the merge. I added a new directory with ARKODE specific documentation in the full RHS PR.

File: doc/superbuild/source/developers/style_guide/index.rst [hunk 2]
```diff
@@ -14,15 +14,14 @@
 
 .. _Style:
 
-***********
```
Comment (gardner48) [inline doc/superbuild/source/developers/style_guide/index.rst:17]: Add `===========` above (for some reason we can't suggest changes to a deleted line)

File: doc/superbuild/source/developers/testing/Answers.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  #. Update the relevant ``.out`` files in
+     ``test/answers/linux-ubuntu20.04-x86_64/gcc-9.4.0/<double|single|extended>`` (this is the files
+     used for the GitHub Actions CI). Like with the embedded `.out` files, you can try and use the
+     output generated on your machine, but it may be different enough from what is produced on the
+     GitHub actions to trigger a failure. You can download the output files generated on the GitHub
+     machines by going to `<https://github.com/LLNL/sundials/actions>`_ finding your failing
+     test, clicking it, then at the bottom downloading the "output_files" artifact (you can also find your
+     failing test at the bottom of a PR). The downloaded zip file will be the SUNDIALS build
+     directory. The ``.out`` files you want are in
+     ``build_that_you_downloaded_and_extracted/Testing/output``.
```
Comment (gardner48) [inline doc/superbuild/source/developers/testing/Answers.rst:70]: To update just the failed files you can use `scripts/updateOutFiles.py` e.g.,
```
cd scripts
./updateOutFiles.py <source path> <destination path>
```
When updating `test/answers` the source path is the build directory downloaded from GitHub and the destination path is  `test/answers/linux-ubuntu20.04-x86_64/gcc-9.4.0/<double|single|extended>`.

When updating with output from Jenkins the source path is the build directory for the Jenkins run and the destination path is the top level of the sundials repo (i.e., `../.` if running from `scripts`). The scripts will automatically traverse the `examples` and `test/unit_test` output files and update files for the failed tests.

File: doc/superbuild/source/developers/testing/Answers.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+- For the ``.out`` files in `sundials-codes/answers <https://github.com/sundials-codes/answers>`_:
+
+  #. Navigate to `test/answers` in the sundials repository. If there is nothing in this directory,
+     you need to run `git submodule init && git submodule update`.
+
+  #. Checkout the main branch: ``git checkout main``. 
+
+  #. Create a new branch off of main. It is ideal if you use a branch name that matches the name of
+     the branch that you are developing in sundials on: ``git checkout -b <branch name>``.
```
Comment (gardner48) [inline doc/superbuild/source/developers/testing/Answers.rst:59]: Does this work from within the submodule? Looks like the remote is an `https` address so I'd need to setup an [alternative authentication method](https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls). 

Usually I clone the `answers` repo separately push a new branch from there and update the commit in `test/answers`

File: doc/superbuild/source/developers/testing/Answers.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  tests running on Jenkins still fail, then you will need to download the output file generated on
+  the Jenkins CI machine and commit it. Extracting this file from the Jenkins machine may require
+  help from David or Cody currently.
+
+- For the ``.out`` files in `sundials-codes/answers <https://github.com/sundials-codes/answers>`_:
+
+  #. To add new answers, start by copying the relevant answers in ``linux-ubuntu20.04-x86_64`` to
+     your new directory (following the naming scheme outlined above).
+
+  #. Next Commit these files and open a PR to the "staging" branch.
```
Comment (gardner48) [inline doc/superbuild/source/developers/testing/Answers.rst:89]: Is the staging branch necessary? Recently we've been adding new answer files through a PR to `main`.

**Review by gardner48 — CHANGES_REQUESTED** [review]

File: doc/superbuild/source/developers/testing/Answers.rst (hunk 4 above)
Comment (balos1) [reply doc/superbuild/source/developers/testing/Answers.rst:59]: You just have to add a different remote from within the submodule. Ill add some more details about both.

File: doc/superbuild/source/developers/testing/Answers.rst (hunk 5 above)
Comment (balos1) [reply doc/superbuild/source/developers/testing/Answers.rst:89]: By new answers, I meant copies of the answers but updated for a different machine. In that case we need to staging branch to be able to diff the out files. Ill clarify in the text.

**Comment by drreynolds** [comment]
Do the docs in each PR get built automatically?  If so, could we add the ReadTheDocs URL for this PR as a comment here?

**Comment by balos1** [comment]
Yes, for this branch I told RTD to build the docs automatically. Here is the link https://sundials.readthedocs.io/en/feature-udpating-answers-instructions/

**Comment by balos1** [comment]
GitLab CI is reporting as failed since Corona is (or was) down. I will rerun the jobs. It would be nice to merge this PR before the removal of deprecated code PR @gardner48.

File: doc/superbuild/source/developers/testing/Answers.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+     path is the top level of the sundials repo (i.e., ../. if running from scripts). The script
+     will automatically traverse the examples and test/unit_test directories to update output files
+     for failed tests.
+
+  #. If you need to use ssh authentication for pushing to GitHub, then you may need to update the 
+     remote for the submodule:
+
+   .. code::
+
+         git remote set-url origin git@github.com:LLNL/sundials.git
```
Comment (gardner48) [inline doc/superbuild/source/developers/testing/Answers.rst:90]: Is this the right remote? Or should it be sundials-codes/answers

**Review by gardner48 — APPROVED** [review]
Minor question about the submodule remote but otherwise this looks good.

We'll need to make some updates for displaying graphviz figures on the readthedocs page (or it seems that way on my phone) but that's a separate issue.

File: doc/superbuild/source/developers/testing/Answers.rst (hunk 6 above)
Comment (balos1) [reply doc/superbuild/source/developers/testing/Answers.rst:90]: Good catch, it should be sundials-codes/answers.

Comment (balos1) [reply doc/superbuild/source/developers/testing/Answers.rst:90]:
```suggestion
         git remote set-url origin git@github.com:sundials-codes/answers.git
```

**Review by gardner48 — APPROVED** [review]

**Pull request merged**